local/local.go

```diff
@@ -351,10 +354,17 @@ func (i *Image) doSave() (types.ImageInspect, error) {
 			done <- err
 			return
 		}
-		if err := ensureReaderClosed(res.Body); err != nil {
-			done <- errors.Wrap(err, "failed to drain and close response from docker client")
+
+		//only return response error after response is drained and closed
+		responseErr := checkResponseError(res.Body)
+		drainCloseErr := ensureReaderClosed(res.Body)
+		if responseErr != nil {
+			done <- responseErr
 			return
 		}
+		if drainCloseErr != nil {
+			done <- drainCloseErr
+		}
```

Comment:
Ok, here's another attempt. Is it ok that we're explicitly prioritizing response error over the close error? My feeling is that the response errors would be important but I haven't seen many examples of `io.Copy()` or `r.Close()` errors.

Reply:
I think prioritizing the response error is the right call.